src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -105,6 +105,7 @@ static List * CreateSubqueryTargetListAndAdjustVars(List *columnList);
 static AttrNumber FindResnoForVarInTargetList(List *targetList, int varno, int varattno);
 static bool RelationInfoContainsOnlyRecurringTuples(PlannerInfo *plannerInfo,
 													Relids relids);
+static char * RecurringTypeDescription(RecurringTuplesType recurType);
```

Comment:
should we get this change to main with a separate PR?  This PR is already pretty large, would be great to extract out as much as possible (if this doesn't require too much effort on your side)

Reply:
This would probably mean fixing a lot of error messages on main first, and then fixing them again on this branch, so I'm more inclined to do that here.